jdaviz/app.py

```diff
@@ -1168,6 +1168,12 @@ def _update_selected_data_items(self, viewer_id, selected_items):
                     and active_data._preferred_translation is not None):
                 self._set_plot_axes_labels(viewer_id)
 
+            # Make everything visible again in Imviz.
+            if self.config == 'imviz':
+                for lyr in viewer.state.layers:
+                    lyr.visible = True
+                viewer.on_limits_change(only_if_open=True)  # Trigger compass redraw
+
```

Comment:
This does seem to work for the bug, but also has the consequence of re-enabling any intentionally disabled layers every time data is checked/unchecked/added.  This might be the best incremental solution for now, as I think the only way to avoid this would be to handle tracking the blinking state separately from the visible state.

Reply:
I'm still really torn about this.  Imviz supports both blinking and viewing multiple semi-transparent layers at once, but switching between those two "modes of operation" will still result in unexpected behavior.  That of course will take a lot more thought to fully "fix", and so I do think this is helpful for now.

Should we at least include a check that the layers we're forcing to be visible are data-layers and not regions, etc?  That's probably an edge case where a user would intentionally set a subset to be hidden, but at least this hard reset would affect as few layers as possible then.